### PR TITLE
http3 transport for miniooni

### PR DESCRIPTION
**Pull request by kelmenhorst**

### CODE REVIEW

**Review by bassosimone — CHANGES_REQUESTED** [review]
Thank you so much! This is very good work! I have requested a bunch of style changes and have pointed out that we need to sync up this branch with changes in OONI's master before we can proceed and merge the code. After this sync has occurred, we should also add a new test case that runs Go 1.15 along with Go 1.14 to have more confidence both cases are working. This can be done by creating a matrix in `workflows/shortttests.yml`.

Thanks again! 🎸

File: netx/httptransport/http3transport.go [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+// NewHTTP3Transport creates a new http3 transport.
+// That is a transport using the quic-go library.
+func NewHTTP3Transport(dialer Dialer, tlsDialer TLSDialer) RoundTripper {
+	txp := &HTTP3Transport{}
+	txp.QuicConfig = &quic.Config{}
+	txp.Dial = func(network, addr string, tlsCfg *tls.Config, cfg *quic.Config) (quic.EarlySession, error) {
+		// quic.DialAddrEarlyContext is not in the official release of quic-go yet
+		return quic.DialAddrEarly(addr, tlsCfg, cfg)
+	}
```
[the remark this answers is not in the document]
Comment (kelmenhorst) [reply netx/httptransport/http3transport.go:32]: Correct. This was just to show that it is possible to use custom dialers here. But you're right, practically, this is not useful currently.

File: libminiooni/libminiooni.go [hunk 2]
```diff
@@ -398,16 +398,18 @@ func MainWithConfiguration(experimentName string, currentOptions Options) {
 		log.Infof("Report ID: %s", experiment.ReportID())
 	}
 
+	usingTransport := map[string]string{
+		"false": " over TCP",
+		"true":  " over QUIC",
+	}
```
Comment (bassosimone) [inline libminiooni/libminiooni.go:404]: Nice!

File: netx/httptransport/http3transport.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (bassosimone) [reply netx/httptransport/http3transport.go]: Yeah, I think it has been useful to understand how to proceed and we can easily re-add later! Thanks!

**Review by bassosimone — APPROVED** [review]
Thank you so much @kelmenhorst, you have done an excellent work here! Looking forward to continue and experiment with `h3`. I'm now going to merge this branch!

🐳

**Pull request merged**